pulpcore/tests/functional/api/using_plugin/test_repo_versions.py

```diff
@@ -532,3 +533,113 @@ def test_all(self):
                 last_task['created_resources'][0],
                 last_task['created_resources']
             )
+
+
+class CreateRepoBaseVersionTestCase(unittest.TestCase):
+    """Test whether one can create a repository version from any version.
+
+    This test targets the following issues:
+
+    `Pulp #3360 <https://pulp.plan.io/issues/3360>`_
+    `Pulp #4035 <https://pulp.plan.io/issues/4035>`_
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables."""
+        cls.cfg = config.get_config()
+        cls.client = api.Client(cls.cfg, api.json_handler)
+        populate_pulp(cls.cfg, url=FILE_LARGE_FIXTURE_MANIFEST_URL)
+        cls.content = sample(cls.client.get(FILE_CONTENT_PATH)['results'], 10)
+
+    def test_same_repository(self):
+        """Test ``base_version`` for the same repository.
+
+        Do the following:
+
+        1. Create a repository.
+        2. Sync the repository (this creates repository version 1).
+        3. Add a new content unit a new repository version (this create
+           repository version 2).
+        4. Create a new repository version using version 1 as ``base_version``
+           (this creates version 3).
+        5. Check that version 1 and version 3 have the same content.
+        """
+        repo = self.create_sync_repo()
+        added_content = []
+        added_content.append(get_added_content(repo))
+        self.assertIsNone(get_versions(repo)[0]['base_version'])
+
+        content = self.content.pop()
+        self.client.post(
+            repo['_versions_href'],
+            {'add_content_units': [content['_href']]}
+        )
+        repo = self.client.get(repo['_href'])
+
+        self.client.post(
+            repo['_versions_href'],
+            {'base_version': get_versions(repo)[0]['_href']}
+        )
+        repo = self.client.get(repo['_href'])
+        added_content.append(get_content(repo))
+
+        self.assertEqual(added_content[0], added_content[1], added_content)
```

Comment:
This test was hard to read. I think part of the reason is you are comparing added_content from repo version 1 to the content of repo version 2. I would recommend not dealing with added content and just examine the content of both repo versions. Something like:

```python
        # create repo version 1
        repo = self.create_sync_repo()
        version_1_content = get_content(repo)
        self.assertIsNone(get_versions(repo)[0]['base_version'])

        # create repo version 2
        self.client.post(
            repo['_versions_href'],
            {'remove_content_units': [version_1_content[0]['_href']]}
        )
        repo = self.client.get(repo['_href'])

       # create repo version 3 from version 1
        self.client.post(
            repo['_versions_href'],
            {'base_version': get_versions(repo)[0]['_href']}
        )
        repo = self.client.get(repo['_href'])
        version_3_content = get_content(repo)

        self.assertEqual(version_1_content, version_3_content)
````